lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedAccessControlModule.java

```diff
@@ -15,43 +15,79 @@
-import com.google.inject.Module;
 import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import com.google.inject.TypeLiteral;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
 import io.airlift.log.Logger;
+import io.airlift.units.Duration;
 import io.trino.plugin.base.CatalogName;
 import io.trino.spi.connector.ConnectorAccessControl;
 
-import java.io.File;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
 
 import static com.google.common.base.Suppliers.memoizeWithExpiration;
+import static io.airlift.configuration.ConditionalModule.conditionalModule;
 import static io.airlift.configuration.ConfigBinder.configBinder;
+import static io.airlift.http.client.HttpClientBinder.httpClientBinder;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 public class FileBasedAccessControlModule
-        implements Module
+        extends AbstractConfigurationAwareModule
 {
     private static final Logger log = Logger.get(FileBasedAccessControlModule.class);
 
     @Override
-    public void configure(Binder binder)
+    public void setup(Binder binder)
     {
+        bindRules();
         configBinder(binder).bindConfig(FileBasedAccessControlConfig.class);
     }
 
+    private void bindRules()
+    {
+        FileBasedAccessControlConfig configuration = buildConfigObject(FileBasedAccessControlConfig.class);
+        install(conditionalModule(
+                FileBasedAccessControlConfig.class,
+                config -> config.isRest(),
+                innerBinder -> {
+                    innerBinder.bind(new TypeLiteral<Supplier<AccessControlRules>>() {})
+                            .to(RestFileBasedAccessControlRulesProvider.class)
+                            .in(Scopes.SINGLETON);
+                    httpClientBinder(innerBinder).bindHttpClient("security-http-client", ForAccessControlRules.class)
+                            .withConfigDefaults(config -> config
+                                    .setRequestTimeout(Duration.succinctDuration(10, TimeUnit.SECONDS))
+                                    .setSelectorCount(1)
+                                    .setMinThreads(1));
+                }));
+        install(conditionalModule(
+                FileBasedAccessControlConfig.class,
+                config -> !config.isRest(),
+                innerBinder -> {
+                    innerBinder.bind(new TypeLiteral<Supplier<AccessControlRules>>() {})
+                            .toProvider(() -> new LocalFileAccessControlRulesProvider<>(configuration, AccessControlRules.class))
+                            .in(Scopes.SINGLETON);
+                }));
+    }
+
     @Inject
     @Provides
-    public ConnectorAccessControl getConnectorAccessControl(CatalogName catalogName, FileBasedAccessControlConfig config)
+    public ConnectorAccessControl getConnectorAccessControl(
+            CatalogName catalogName,
+            FileBasedAccessControlConfig config,
```

Comment:
Since it is common to both config I guess we can extract them as a different config ?

Reply:
we have different configs now